ios/MaasCore/Sources/MaasComponents/Components/BottomSheet/BottomSheet.swift

```diff
@@ -0,0 +1,140 @@
+import SwiftUI
+import Combine
+
+struct BottomSheet: ViewModifier {
+
+    @Themeable(BottomSheetConstants.init) var constants
+
+    @EnvironmentObject var environment: BottomSheetEnvironment
+
+    @State var dragOffset: CGFloat = 0
+
+    @State var contentBounds: CGRect = .zero
+
+    @State var presenterBounds: CGRect = .zero
+
+    var topAnchor: CGFloat { presenterBounds.height - contentBounds.height }
```

Comment:
Is it valid for this to be negative? If the content is taller than the presenter.

Reply:
If you define a content taller then presenter you should be aware that if content is not in scrollView some of your content will be hidden beneath device bottom. 😄 

Attaching full screen bottom sheet. On top you can see `safeAreaInsets` we could discard this, but at the moment it feels right for bottom sheet to have this than crazy animated header implementation...

![Simulator Screen Shot - iPhone 12 Pro Max - 2021-02-15 at 11 41 13](https://user-images.githubusercontent.com/51905686/107929707-b6e7ac80-6f82-11eb-90e3-20f0a4eb3179.png)